### PR TITLE
Postpone buckets for reconsideration until it is safe.

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp [hunk 1]
```diff
@@ -53,6 +53,8 @@ blockedDueToClusterState(const std::shared_ptr<IBucketStateCalculator> &calc)
     return !(clusterUp && nodeUp && !nodeInitializing);
 }
 
+constexpr BucketId RECOMPUTE_BUCKETID;
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp:56]: Above name indicates that the scope is a single bucket.  But the usage indicates that the scope is all buckets.

File: searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp [hunk 2]
```diff
@@ -234,34 +242,59 @@ BucketMoveJobV2::prepareMove(BucketMoverSP mover, std::vector<MoveKey> keys, IDe
...
+BucketMoveJobV2::handleMoveResult(BucketMoverSP mover) {
     if (mover->bucketDone() && mover->inSync()) {
-        _modifiedHandler.notifyBucketModified(mover->getBucket());
+        BucketId bucket = mover->getBucket();
+        assert(_bucketsInFlight.contains(bucket));
+        _modifiedHandler.notifyBucketModified(bucket);
+        _bucketsInFlight.erase(bucket);
+        updatePending();
+        if (_postponedUntilSafe.contains(bucket)) {
+            reconsiderBucket(_ready.meta_store()->getBucketDB().takeGuard(), bucket);
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp:257]: Missing a `_postponedUntilSafe.erase(bucket)` before reconsiderBucket ?

File: searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp]: Agree, RECOMPUTE_TOKEN it is.

File: searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp (hunk 2 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp:258]: Good point eternal work is never good.

**Review by geirst — COMMENTED** [review]
@baldersheim Is this something that should be tested in a unit test?

**Pull request merged**